robot/packages/cloud_package_manager.go

```diff
@@ -299,23 +301,30 @@ func sanitizeURLForLogs(u string) string {
 	return parsed.String()
 }
 
-func (m *cloudManager) loadFile(ctx context.Context, url string, p config.PackageConfig) error {
+func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config.PackageConfig) error {
 	// TODO(): validate integrity of directory.
-	if dirExists(m.localDataPath(p)) {
+	if dirExists(p.LocalDataDirectory(m.packagesDir)) {
 		m.logger.Debug("  Package already downloaded, skipping.")
 		return nil
 	}
 
+	if err := os.MkdirAll(p.LocalDataParentDirectory(m.packagesDir), 0o700); err != nil {
```

Comment:
nit: i feel like we should do this in dirExists function as that seems more like a validation check.

Reply:
Agreed. This function is causing tests to fail

Reply:
Upon second reading of your comment, I suspect this comment was written before a few other refactors. 

This is not a validation check, this is creating the directory that we need to download data into (ex: packages/.data/module) I would be surprised if  `dirExists` created a directory if the dir didn't exist... then `dirExists` would be a `createDirIfNotExist`, but that is just `MkdirAll`

Reply:
No yeah I understand its not a validation check, but I thought it may make sense to do that directory creation / validation stuff in a separate function just to break up the length of this large function. 

Not a big deal though.

Reply:
In favor of keeping it here for now because this is the pattern followed for the `.data` dir and this seems like a reasonable path forwards. Agreed that it is not perfect, but at least it is clear to the reader when the dir is created. Ehh.. idk